src/bin/bts_decode.py

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env python
+
+# Copyright 2018 Nagoya University (Tomoki Hayashi)
+#  Apache 2.0  (http://www.apache.org/licenses/LICENSE-2.0)
+
+import argparse
+import logging
+import os
+import subprocess
+import sys
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    # general configuration
+    parser.add_argument('--gpu', default=None, type=int, nargs='?',
+                        help='GPU ID (negative value indicates CPU)')
+    parser.add_argument('--ngpu', default=0, type=int,
+                        help='Number of GPUs')
+    parser.add_argument('--backend', default='pytorch', type=str,
+                        choices=['chainer', 'pytorch'],
+                        help='Backend library')
+    parser.add_argument('--out', type=str, required=True,
+                        help='Output filename')
+    parser.add_argument('--verbose', '-V', default=0, type=int,
+                        help='Verbose option')
+    # task related
+    parser.add_argument('--json', type=str, required=True,
+                        help='Filename of train label data (json)')
+    parser.add_argument('--model', type=str, required=True,
+                        help='Model file parameters to read')
+    parser.add_argument('--model-conf', type=str, required=True,
+                        help='Model config file')
+    # decoding related
+    parser.add_argument('--maxlenratio', type=float, default=5,
+                        help='Maximum length ratio in decoding')
+    parser.add_argument('--minlenratio', type=float, default=0,
+                        help='Minimum length ratio in decoding')
+    parser.add_argument('--threshold', type=float, default=0.5,
+                        help='Threshold value in decoding')
+    args = parser.parse_args()
+
+    # logging info
+    if args.verbose > 0:
+        logging.basicConfig(
+            level=logging.INFO, format='%(asctime)s (%(module)s:%(lineno)d) %(levelname)s: %(message)s')
+    else:
+        logging.basicConfig(
+            level=logging.WARN, format='%(asctime)s (%(module)s:%(lineno)d) %(levelname)s: %(message)s')
+        logging.warning('Skip DEBUG/INFO messages')
+
+    # check gpu argument
+    if args.gpu is not None:
+        logging.warn("--gpu option will be deprecated, please use --ngpu option.")
+        if args.gpu == -1:
+            args.ngpu = 0
+        else:
+            args.ngpu = 1
+
+    # check CUDA_VISIBLE_DEVICES
```

Comment:
better to use lines from https://github.com/espnet/espnet/pull/260/files#diff-dbc60e85a2905c1803821c64a4c81cb0R147 to make it use in CLSP